src/polymer/iron-flex-layout-classes.ts

```diff
@@ -158,31 +159,44 @@ ${indent}</custom-style>`,
   }
 }
 
-function getMissingStyleModules(node: dom5.Node) {
+function getMissingStyleModules(rootNode: dom5.Node):
+    {module: string, node: dom5.Node}[] {
   let includes = '';
-  for (const style of dom5.queryAll(node, isStyleInclude)) {
-    includes += ' ' + dom5.getAttribute(style, 'include')!;
+  const modules = {};
+  dom5.nodeWalkAll(rootNode, (node: dom5.Node) => {
+    if (dom5.isElement(node)) {
+      if (isStyleInclude(node)) {
+        includes += ' ' + dom5.getAttribute(node, 'include')!;
+      } else {
+        styleModules.forEach((m) => {
+          if (!modules[m.module] && m.selector(node)) {
+            modules[m.module] = node;
```

Comment:
Interesting. Did you consider making modules map to an array of nodes rather than a single one?

On the one hand, it could produce a flood of warnings. On the other hand, that's how people expect linters to work, and in the IDE use case it's much nicer to see all of the places where the classes are used incorrectly.

Reply:
Updated to warn of missing style module include on every node using it 👌